OSSM-8673 Staging and Testing Migration Guides IA/Content (#88399)
openshift/openshift-docs: migrating/ossm-migrating-from-2-to-3-assembly.adoc

// Module included in the following assemblies:
//
// * service-mesh-docs-main/migrating

:_mod-docs-content-type: ASSEMBLY
[id=ossm-migrating-from-2-to-3-assembly]
= Migrating from Service Mesh 2 to Service Mesh 3
include::_attributes/common-attributes.adoc[]
:context: ossm-migrating-from-2-to-3-assembly

toc::[]

The informational content and step-by-step guides contained within this migration section applies only if:

//there may be additional updates prior to Release Day.

* You are an existing {SMProductName} user running {SMProduct} {SMv2Version}.

> * You want to move to {SMProduct} 3.0.

Reviewer: ```suggestion
* You plan to move to {SMProduct} 3.0.
```

> If you are an existing user but you are not running {SMProduct} {SMv2Version} version, you must update before you can continue. Here's how: link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/service_mesh/service-mesh-2-x#ossm-versioning_ossm-upgrade[Upgrading Service Mesh].

Reviewer: ```suggestion
If you are an existing user but you are not running {SMProduct} {SMv2Version} version, you must update Service Mesh before you can continue. For more information, see link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/service_mesh/service-mesh-2-x#ossm-versioning_ossm-upgrade[Upgrading Service Mesh].
```

**This should be placed after the first prereq maybe as a note**.

[NOTE]
====
From time to time, you may need to reference, or will be directed to {SMProduct} 2.x content. All {SMProduct} 2.x content is available here: link:https://docs.redhat.com/en/documentation/openshift_container_platform/4.17/html/service_mesh/service-mesh-2-x[OpenShift Service Mesh 2.x]. It may be beneficial to open {SMProduct} 2.x content in a new tab or window for easier reference as you move through the migration guides.
====

//02/04/2024: Per Service Mesh Program Call, there will be a 2.6.6 release with 3.0, so users will need to upgrade to 2.6.6. UPDATE THE VARIABLE IN COMMON_ATTRIBUTES ACCORDINGLY
//it looks like the latest z-stream will be 2.6.5, set to release on Feb 5, 2025. Unless there is a z-stream in Feb, 2.6.5 is likely the version to be listed for 2.6.z.

[id="migration-recommendations_{context}"]
== Migration recommendations

> Given the nature of migrating to a brand new operator, the following recommendations are provided to help minimize the risk of misconfigurations or possible conflicts:

Reviewer: ```suggestion
Review the following recommendations to minimize the risk of misconfigurations or possible conflicts:
```
